payment.py

```diff
@@ -289,14 +289,36 @@ def _check_payment(cls, payments):
                         party=payment.party.rec_name,
                         amount=payment.amount,
                         ))
-            elif payment.party not in payment.bank_account.owners:
-                raise UserError(gettext(
-                        'account_payment_sepa_es.'
-                        'msg_bad_relation_party_bank_account',
-                        party=payment.party.rec_name,
-                        amount=payment.amount,
-                        account=payment.bank_account.rec_name,
-                        ))
+            elif payment.journal.payment_type:
+                if (payment.journal.payment_type.account_bank == 'party'
+                        and payment.party not in payment.bank_account.owners):
+                    raise UserError(gettext(
+                            'account_payment_sepa_es.'
+                            'msg_bad_relation_party_bank_account',
+                            party=payment.party.rec_name,
+                            amount=payment.amount,
+                            account=payment.bank_account.rec_name,
+                            ))
+                elif (payment.journal.payment_type.account_bank == 'company'
+                        and payment.company.party not in payment.bank_account.owners):
+                    raise UserError(gettext(
+                            'account_payment_sepa_es.'
+                            'msg_bad_relation_party_bank_account',
+                            party=payment.company.party.rec_name,
+                            amount=payment.amount,
+                            account=payment.bank_account.rec_name,
+                            ))
+                elif (payment.journal.payment_type.account_bank == 'other'
+                        and payment.journal.payment_type.party == payment.party
+                        and payment.party not in payment.bank_account.owners):
```

Comment:
The control would be:

              elif (payment.journal.payment_type.account_bank == 'other'
                        and payment.journal.payment_type.party
                        not in payment.bank_account.owners):

Reply:
7742c7246da31dab62829812b4aa07b8f770ce58